R/zchunk_LA1231.elec_tech.R

```diff
@@ -37,59 +38,153 @@ module_energy_LA1231.elec_tech_DISABLED <- function(command, ...) {
+      select(GCAM_region_ID,sector,fuel,technology,year,value) -> L1231.in_EJ_R_elec_F_tech_Yh
+
+    #Calculate output as input * efficiency
+    #This tibble only includes technologies with modeled efficiencies
+    L1231.eff_R_elec_F_tech_Yh %>%
+      left_join(L1231.in_EJ_R_elec_F_tech_Yh,by=c("GCAM_region_ID","sector","fuel","technology","year" )) %>%
+      mutate(output=efficiency*value) %>%
+      select(-value,-efficiency) -> L1231.out_EJ_R_elec_Fin_tech_Yh
+
+    # Rename efficiecy as value for final output in long format
+    L1231.eff_R_elec_F_tech_Yh %>%
+      rename(value=efficiency) -> L1231.eff_R_elec_F_tech_Yh
+
+    # Combine output with technologies modeled by output only (e.g. nuclear, hydro, renewables)
+    # and combined all outpues for all fuels and technologies to create final tibble in long format
+    L123.out_EJ_R_elec_F_Yh %>%
+     anti_join(L1231.out_EJ_R_elec_Fin_tech_Yh, by=c("GCAM_region_ID", "sector","fuel","year")) %>%
+      arrange(fuel,GCAM_region_ID,year) %>%
+      left_join(select(calibrated_techs,sector,fuel,technology),by=c("sector" ,"fuel")) %>%
+      rename(output=value) %>%
+      bind_rows(L1231.out_EJ_R_elec_Fin_tech_Yh) %>%
+      rename(value=output) %>%
+      select(GCAM_region_ID, sector, fuel, technology, year, value) -> L1231.out_EJ_R_elec_F_tech_Yh
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # add final details to tibbles and save them
+    L1231.in_EJ_R_elec_F_tech_Yh%>%
+      add_title("Inputs to electricity by Region / fuel / technology") %>%
+      add_units("EJ") %>%
+      add_comments("Written by LA1231.electech.R") %>%
       add_legacy_name("L1231.in_EJ_R_elec_F_tech_Yh") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
+      add_precursors("energy/A23.globaltech_eff", "energy/calibrated_techs","L123.in_EJ_R_elec_F_Yh","L123.out_EJ_R_elec_F_Yh","L123.eff_R_elec_F_Yh") %>%
       add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
       L1231.in_EJ_R_elec_F_tech_Yh
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L1231.out_EJ_R_elec_F_tech_Yh %>%
+      add_title("Outputs of electricity by by Region / fuel / technology") %>%
+      add_units("EJ") %>%
+      add_comments("Written by LA1231.electech.R") %>%
```

Comment:
Need more than this in comments--a sentence or two about how the data were calculated.